src/OAuth2/AbstractProvider.php

```diff
@@ -86,4 +87,15 @@ protected function parseExpiresIn($body)
     {
         return Arr::get($body, 'expires_in');
     }
+
+    /**
+     * Get the approved scopes from the token response body.
+     *
+     * @param  array  $body
+     * @return array
+     */
+    protected function parseApprovedScopes($body)
+    {
+        return explode($this->scopeSeparator, Arr::get($body, 'scope', ''));
```

Comment:
```suggestion
        $scopesRaw = Arr::get($body, 'scope',null);
        
        if ($scopesRaw === null || (!is_array($scopesRaw) && !is_string($scopesRaw)) {
        	return [];
        	}
        
        if (is_array($scopesRaw)) {
        return $scopesRaw;
        }
        
        return explode($this->scopeSeparator, Arr::get($body, 'scope', ''));
```

Reply:
This covers the following case:

- no scopes set
- array scopes (some providers return an array)
- string scopes

Reply:
This looks good to me 👍 But note that in Socialite the scopes must be a string or `null`, which is then exploded to an array. But you are allowing it to also be an array.

Don't forget to increase the minimum version requirement to `socialiteproviders/manager:^5.5` because the `setApprovedScopes()` function does [not exist prior to this version](https://github.com/laravel/socialite/commit/ed2122dce14cab7f49d73ade43b0ef6072787772).